File: src/main/java/me/ag2s/epublib/domain/SpineReference.java
package me.ag2s.epublib.domain;

import java.io.Serializable;

public class SpineReference extends ResourceReference implements Serializable {
   private static final long serialVersionUID = -7921609197351510248L;
   private boolean linear;

   public SpineReference(Resource resource) {
      this(resource, true);
   }

   public SpineReference(Resource resource, boolean linear) {
      super(resource);
      this.linear = linear;
   }

   public boolean isLinear() {
      return this.linear;
   }

   public void setLinear(boolean linear) {
      this.linear = linear;
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/SpineReference.kt
package me.ag2s.epublib.domain

import java.io.Serializable

class SpineReference(resource: Resource?, private var linear: Boolean = true) : ResourceReference(resource), Serializable {
    fun isLinear(): Boolean = linear
    fun setLinear(linear: Boolean) { this.linear = linear }
    companion object { private const val serialVersionUID = -7921609197351510248L }
}
